ec2deprecateimg

```diff
@@ -43,9 +43,18 @@ argparse.add_argument(
     help='AWS access key (Optional)',
     metavar='AWS_ACCESS_KEY'
 )
+help_msg = 'The deprecation date, the date the image is considered '
+help_msg += 'deprecated. The default value is today\'s date (Optional)'
+argparse.add_argument(
+    '--deprecation-date',
+    default='',
+    dest='depDate',
+    help=help_msg,
+    metavar='DEPRECATION_DATE'
+)
```

Comment:
Need to insert a check somewhere in the code that the user provided string is in the correct format, i.e. 8 digits long.

Reply:
I changed the parameter validation code to the argparse logic in the script (makes more sense to fail fast if the parameter does not have the proper format, right?)

Also relaxed the logics in the lib functions, as the parameter is validated in advance.

Reply:
Maybe for a different PR but this could be a configuration setting that would allow user to choose their own date format.